kratos/utilities/variable_utils.cpp

```diff
@@ -323,6 +323,24 @@ KRATOS_API(KRATOS_CORE) const ModelPart::ConditionsContainerType& VariableUtils:
     return rModelPart.Conditions();
 }
 
+template<class TDataType>
+void VariableUtils::AuxiliaryHistoricalValueSetter(
+    const Variable<TDataType>& rVariable,
+    const TDataType& rValue,
+    NodeType& rNode)
+{
+    rNode.FastGetSolutionStepValue(rVariable) = rValue;
+}
+
+template<>
+void VariableUtils::AuxiliaryHistoricalValueSetter(
+    const Variable<array_1d<double,3>>& rVariable,
+    const array_1d<double,3>& rValue,
+    NodeType& rNode)
+{
+    noalias(rNode.FastGetSolutionStepValue(rVariable)) = rValue;
```

Comment:
Does ```noalias``` make a difference in a simple assignment?

Reply:
Agree, should the `noalias` be in the above function?

Reply:
> Does `noalias` make a difference in a simple assignment?

It does.

Reply:
> Agree, shouldn't the `noalias` be in the above function?

Nope, as it doesn't work with all variable types. This is why I needed this auxiliary function.

Reply:
I don't understand, afaik `noalias` is used to prevent operations like `A = A + B`, from being executed as `temp = A + B, A = temp` (and 2nd+ level pointer indirections but we don't have variables like those don't we?).

Setting a `rValue` is already that operation. The only scenario in which this could fail, is if you do something like this:

```c++
// This would forcefully modify MATRIX_VAR while calculating it making it yield and incorrect value during the op
AuxiliaryHistoricalValueSetter(MATRIX_VAR, myNode.FastGetSolutionStepValue(MATRIX_VAR) * myNode.FastGetSolutionStepValue(MATRIX_VAR_2), myNode)
```

Am I missing something? Which is the case or the type of the var that is making no-alias behave differently?

Reply:
More than behaving differently, the  `noalias` doesn't work with `double` type variables. This is why we need the auxiliary function. I'm not aware about the technicalities behind, for this we'd better ask @RiccardoRossi .